### PR TITLE
add crud for organisations

### PR DESCRIPTION
**Pull request by danvixent**

### CODE REVIEW

File: datastore/mongo/organisation.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+		return err
+	}
+
+	return nil
+}
+
+func (db *orgRepo) FetchOrganisationByID(ctx context.Context, id string) (*datastore.Organisation, error) {
+	org := new(datastore.Organisation)
+
+	err := db.inner.FindOne(ctx, bson.M{"uid": id}).Decode(&org)
```
Comment (Dotunj) [inline datastore/mongo/organisation.go:75]: We can scope the query to the `document_status` as well, so organisations that have been deleted are not returned.

Comment (danvixent) [reply datastore/mongo/organisation.go:75]: Yes, this is a known issue.

File: server/route.go [hunk 2]
```diff
@@ -172,6 +172,21 @@ func buildRoutes(app *applicationHandler) http.Handler {
 				})
 			})
 
+			r.Route("/organisations", func(orgRouter chi.Router) {
```
Comment (Subomi) [inline server/route.go:175]: @danvixent This should be under the `/ui` router.

File: server/organisation_integration_test.go [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+	var organisation datastore.Organisation
+	parseResponse(s.T(), w.Result(), &organisation)
+
+	org, err := s.DB.OrganisationRepo().FetchOrganisationByID(context.Background(), uid)
+	require.NoError(s.T(), err)
+	require.Equal(s.T(), "new_org", org.Name)
+	require.Equal(s.T(), "new_org", organisation.Name)
+}
+
+func (s *OrganisationIntegrationTestSuite) Test_GetOrganisations() {
```
Comment (Subomi) [inline server/organisation_integration_test.go:165]: @danvixent Can we refactor this to look like this - https://github.com/frain-dev/convoy/blob/b377ed70bb7ccb8af69c67c0d4c66fc3a6de1321/server/application_integration_test.go#L429

File: services/app_service_test.go [hunk 4]
```diff
@@ -791,6 +791,10 @@ func stripVariableFields(t *testing.T, obj string, v interface{}) {
 		a := v.(*datastore.APIKey)
 		a.UID, a.MaskID, a.Salt, a.Hash = "", "", "", ""
 		a.CreatedAt, a.UpdatedAt = 0, 0
+	case "organisation":
```
Comment (Subomi) [inline services/app_service_test.go:794]: Do we still need this? @danvixent

**Review by Subomi — CHANGES_REQUESTED** [review]
@danvixent FYI

File: services/app_service_test.go (hunk 4 above)
Comment (danvixent) [reply services/app_service_test.go:794]: yes

File: server/organisation_integration_test.go (hunk 3 above)
Comment (danvixent) [reply server/organisation_integration_test.go:165]: @Subomi i'm not sure i see the difference here.

File: datastore/mongo/organisation_test.go [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+		err := orgRepo.CreateOrganisation(context.Background(), org)
+		require.NoError(t, err)
+	}
+
+	organisations, _, err := orgRepo.LoadOrganisationsPaged(context.Background(), datastore.Pageable{
+		Page:    2,
+		PerPage: 2,
+		Sort:    -1,
+	})
+	fmt.Println("organisations", organisations)
```
[the remark this answers is not in the document]
Comment (danvixent) [reply datastore/mongo/organisation_test.go:42]: Yeah. Thanks!

File: server/organisation_integration_test.go (hunk 3 above)
Comment (Subomi) [reply server/organisation_integration_test.go:165]: The latter uses `testdb.SeedMultipleEndpoints` to seed multiple endpoints at once that use a randomly generated number of organisations. @danvixent

Comment (Subomi) [reply server/organisation_integration_test.go:165]: We need something like `testdb.SeedMultipleOrganisations`

File: services/app_service_test.go (hunk 4 above)
Comment (Subomi) [reply services/app_service_test.go:794]: Where? @danvixent

Comment (danvixent) [reply services/app_service_test.go:794]: L97 organisation_service_test.go

File: server/organisation_integration_test.go (hunk 3 above)
Comment (danvixent) [reply server/organisation_integration_test.go:165]: Noted

File: datastore/repository.go [hunk 6]
```diff
@@ -48,6 +48,14 @@ type GroupRepository interface {
 	FillGroupsStatistics(ctx context.Context, groups []*Group) error
 }
 
+type OrganisationRepository interface {
```
[the remark this answers is not in the document]
Comment (danvixent) [reply datastore/repository.go:51]: okay

File: server/route.go (hunk 2 above)
Comment (danvixent) [reply server/route.go]: fixed

File: services/app_service_test.go (hunk 4 above)
Comment (Subomi) [reply services/app_service_test.go:794]: 👍🏽

**Review by Subomi — APPROVED** [review]

**Pull request merged**